### DISCUSSION TITLE
Direct-xip support in bootutil_public module

### DISCUSSION THREAD

**Discussion by ghost** [mcu-tools/mcuboot#1474] [Q&A]
Hi

We are converting projects originally using swap mechanism to direct-xip mode because we want to use flash remapping feature on some processors. Unfortunately it seems that bootutil_public.c supports only swap mechanism and is not direct-xip mode aware.
Is there another recommended way or it will be updated in future releases?

**Comment by nvlsianpu** [comment]
Hi. What is missing in bootutil_public.c? What port are you using (zepyr-rtos)?
Adding proper support to the bootutil_public shouldn't be a big task.

**Reply by ghost** [reply]
User needs read, sets and erase image trailers of the specific slot when direct-xip is used.
At the moment these functions in bootutil_public.c are usable only for swap mechanism so they have hardcoded primary slot for active image and secondary slot for candidate image.
I think there should be direct-xip variants - with another argument what specifies the slot to operates on.

**Reply by nvlsianpu** [reply]
@DanielStindl-NXP We know that - at some point we were working on that application will recognize which slot it should thread "how". This was blocked du the disagreement in the community (caused by missing multi-image-build in the zephyr-rtos build system). 
Now we have the sysbuild. I think now it would be possible (easily) to add feature of build the app for both slots and redirect bootutil_public accordingly.

**Reply by ghost** [reply]
@nvlsianpu That would be great :)
Thank you

**Comment by markuckermann** [comment]
Is there work being done on this? I'm trying to implement direct-xip and think this would be useful.
